framework/src/executor/mod.rs

```diff
@@ -217,62 +226,86 @@ impl<S: 'static + Storage, DB: 'static + TrieDB, Mapping: 'static + ServiceMappi
         Ok(ServiceContext::new(ctx_params))
     }
 
-    fn catch_call(&self, context: ServiceContext, exec_type: ExecType) -> ExecResp {
-        let res = match exec_type {
-            ExecType::Read => self.call(context, exec_type),
-            ExecType::Write => self.call_with_tx_hooks(context, exec_type),
-        };
-        match res {
-            Ok(resp) => resp,
-            Err(e) => ExecResp {
-                ret:      e.to_string(),
-                is_error: true,
-            },
+    fn catch_call(
+        &mut self,
+        context: ServiceContext,
+        exec_type: ExecType,
+    ) -> ProtocolResult<ServiceResponse<String>> {
```

Comment:
Remove this retry logic, executor  shouldn't care retry.

Reply:
fixed